src/color.c

```diff
@@ -165,7 +165,7 @@ int image_has_transparency (const struct plum_image * image) {
         while (count --) if (*(color ++) >= 0x7fff) return 1;
       return 0;
     }
-    case PLUM_COLOR_32X: {
+    case PLUM_COLOR_32X: default: {
```

Comment:
This function has intentional UB when the value is not one of the cases because the optimizer doesn't pick it up otherwise. (If it did, you wouldn't be getting a warning.)
I'd add `__builtin_unreachable();` at the end, but I've purposely stayed away from compiler-specific things, even under `#ifdef` guards.

Reply:
Another alternative would be `return 0; // unreachable`.